build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -0,0 +1,241 @@
+
+            // update dotnetcore sdks and runtimes
+            foreach (var platformConstant in platformConstants)
+            {
+                string version = platformConstant.Version;
+                string sha = platformConstant.Sha;
+                string versionType = platformConstant.VersionType;
+                string dotNetConstantKey = GenerateDotNetConstantKey(platformConstant);
+                Console.WriteLine($"[UpdateConstants] version: {version} versionType: {versionType} sha: {sha} dotNetConstantKey: {dotNetConstantKey}");
+                if (versionType.Equals(Constants.SdkName))
+                {
+                    Constant dotNetYamlConstant = dotnetYamlConstants[Constants.DotNetSdkKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // add sdk to versionsToBuild.txt
+                    UpdateVersionsToBuildTxt(platformConstant);
+                }
+                else
+                {
+                    Constant dotNetYamlConstant = dotnetYamlConstants[Constants.DotNetRuntimeKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // store SHAs for net-core and aspnet-core
+                    dotNetYamlConstant.Constants[$"{dotNetConstantKey}-sha"] = sha;
+                }
+            }
+
+            var serializer = new SerializerBuilder()
+                .WithNamingConvention(UnderscoredNamingConvention.Instance)
+                .Build();
+
+            var constantsYamlAbsolutePath = Path.Combine(this.repoAbsolutePath, "build", Constants.ConstantsYaml);
+            var stringResult = serializer.Serialize(yamlConstants);
+            File.WriteAllText(Constants.ConstantsYaml, stringResult);
+        }
+
+        private static void UpdateVersionsToBuildTxt(PlatformConstant platformConstant)
+        {
+            // TODO: use File.ReadAll*
+            List<string> versionsToBuildTxtFiles = new List<string>()
+            {
+                    "platforms/dotnet/versions/bullseye/versionsToBuild.txt",
+                    "platforms/dotnet/versions/buster/versionsToBuild.txt",
+                    "platforms/dotnet/versions/focal-scm/versionsToBuild.txt",
+                    "platforms/dotnet/versions/stretch/versionsToBuild.txt",
+            };
+            foreach (string versionsToBuildTxtFile in versionsToBuildTxtFiles)
+            {
+                string line = $"\n{platformConstant.Version}, {platformConstant.Sha},";
+                File.AppendAllText(versionsToBuildTxtFile, line);
+
+                // sort
+                Console.WriteLine($"[UpdateVersionsToBuildTxt] Updating {versionsToBuildTxtFile}...");
+                var contents = File.ReadAllLines(versionsToBuildTxtFile);
+                Array.Sort(contents);
+                File.WriteAllLines(versionsToBuildTxtFile, contents.Distinct());
```

Comment:
We need to make sure `versionsToBuildTxtFile` is prepended with `repoAbsolutePath`